torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -5510,9 +5529,6 @@ at::Tensor ProcessGroupNCCL::allocateTensor(
   c10::cuda::CUDACachingAllocator::beginAllocateToPool(
       memPool_->device(), memPool_->id(), [](cudaStream_t) { return true; });
   at::Tensor tensor = at::empty({size}, options);
-  // Also need to ncclCommRegister the pool in case new segments are created;
-  // reregistration of old segments will be ignored
-  registerMemPool(memPool_.get());
```

Comment:
Previously, we had to re-register the MemPool each time we did a new allocation, but now, thanks to the new hooks, we only need to register the MemPool once when it's created.